mods/carts/init.lua

```diff
@@ -0,0 +1,355 @@
+	if tool_capabilities and tool_capabilities.full_punch_interval then
+		punch_interval = tool_capabilities.full_punch_interval
+	end
+	time_from_last_punch = math.min(time_from_last_punch or punch_interval, punch_interval)
+	local f = 3 * (time_from_last_punch / punch_interval)
+
+	self.velocity = vector.multiply(cart_dir, f)
+	self.old_dir = cart_dir
+	self.old_pos = nil
+	self.punched = true
+end
+
+function cart_entity:on_step(dtime)
+	local vel = self.object:getvelocity()
+	local update = {}
+	if self.punched then
+		vel = vector.add(vel, self.velocity)
+		self.object:setvelocity(vel)
+		self.old_dir.y = 0
+	elseif vector.equals(vel, {x=0, y=0, z=0}) then
+		return
+	end
+
+	-- dir:         New moving direction of the cart
+	-- last_switch: Currently pressed L/R key, used to ignore the key on the next rail node
+	local dir, last_switch
+	local pos = self.object:getpos()
+
+	if self.old_pos and not self.punched then
+		local flo_pos = vector.round(pos)
+		local flo_old = vector.round(self.old_pos)
+		if vector.equals(flo_pos, flo_old) then
+			-- Do not check one node multiple times
+			return
+		end
+	end
+
+	local ctrl, player
+
+	-- Get player controls
+	if self.driver then
+		player = minetest.get_player_by_name(self.driver)
+		if player then
+			ctrl = player:get_player_control()
+		end
+	end
+
+	if self.old_pos then
+		-- Detection for "skipping" nodes
+		local expected_pos = vector.add(self.old_pos, self.old_dir)
+		local found_path = carts:pathfinder(
+			pos, expected_pos, self.old_dir, ctrl, self.old_switch, self.railtype
+		)
+
+		if not found_path then
+			-- No rail found: reset back to the expected position
+			pos = expected_pos
+			update.pos = true
+		end
+	end
+
+	local cart_dir = carts:velocity_to_dir(vel)
+	local max_vel = carts.speed_max
+	dir, last_switch = carts:get_rail_direction(
+		pos, cart_dir, ctrl, self.old_switch, self.railtype)
+
+	local new_acc = {x=0, y=0, z=0}
+	if vector.equals(dir, {x=0, y=0, z=0}) then
+		vel = {x=0, y=0, z=0}
+		pos = vector.round(pos)
+		update.pos = true
+		update.vel = true
+	else
+		-- If the direction changed
+		if dir.x ~= 0 and self.old_dir.z ~= 0 then
+			vel.x = dir.x * math.abs(vel.z)
+			vel.z = 0
+			pos.z = math.floor(pos.z + 0.5)
+			update.pos = true
+		end
+		if dir.z ~= 0 and self.old_dir.x ~= 0 then
+			vel.z = dir.z * math.abs(vel.x)
+			vel.x = 0
+			pos.x = math.floor(pos.x + 0.5)
+			update.pos = true
+		end
+		-- Up, down?
+		if dir.y ~= self.old_dir.y then
+			vel.y = dir.y * math.abs(vel.x + vel.z)
+			pos = vector.round(pos)
+			update.pos = true
+		end
+
+		-- Slow down or speed up..
+		local acc = dir.y * -1.8
+
+		local node = minetest.get_node(pos)
+		local railparams = carts.railparams[node.name] or {}
+		-- no need to check for railparams == nil since we always make it exist.
+		local speed_mod = railparams.acceleration
+		if speed_mod and speed_mod ~= 0 then
+			-- Try to make it similar to the original carts mod
+			acc = acc + (speed_mod * 10)
```

Comment:
Why not make registered `railparams.acceleration` "5" directly in the definition, instead of using "0.5" and then multiplying it by 10 on every step?

Also, it's misleading. Someone trying to crunch the numbers for their rail might think 0.5 increases the acceleration in 0.5 m/s2 when it actually increases it in 5 m/s2